packages/dashboard-backend/src/api/dockerConfigApi.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { baseApiPath } from '../constants/config';
+import { api } from '@eclipse-che/common';
+import {
+  dockerConfigSchema,
+  namespacedDockerConfigSchema
+} from '../constants/schemas';
+import { getDevWorkspaceClient } from './helper';
+import { restParams } from '../typings/models';
+import { getSchema } from '../services/helpers';
+import { V1Secret } from '@kubernetes/client-node/dist/gen/model/v1Secret';
+
+const tags = ['dockerconfig'];
+
+const secretName = 'devworkspace-container-registry-dockercfg';
+const secretLabels = { 'controller.devfile.io/devworkspace_pullsecret': 'true' };
+
+export function registerDockerConfigApi(server: FastifyInstance) {
```

Comment:
this file mixes REST stuff with business logic for dockerconfig <-> secret. I would distinguish them, similar what we have with workspaceAPI

Reply:
I have fixed it.